plugins/zmq/src/plugin.cpp

```diff
@@ -0,0 +1,415 @@
+        ctrl.abort(message.error());
+    };
+    return make(std::move(*engine));
+  }
+
+  auto name() const -> std::string override {
+    return "zmq";
+  }
+
+  auto default_parser() const -> std::string override {
+    return "json";
+  }
+
+  friend auto inspect(auto& f, zmq_loader& x) -> bool {
+    return f.object(x)
+      .pretty_name("zmq_loader")
+      .fields(f.field("args", x.args_));
+  }
+
+  auto to_string() const -> std::string override {
+    auto result = name();
+    result += fmt::format(" {}", args_.endpoint);
+    if (args_.bind)
+      result += " --bind";
+    if (args_.connect)
+      result += " --connect";
+    return result;
+  }
+
+private:
+  connector_args args_;
+};
+
+class zmq_saver final : public plugin_saver {
+public:
+  zmq_saver() = default;
+
+  zmq_saver(connector_args args) : args_{std::move(args)} {
+  }
+
+  auto instantiate(operator_control_plane& ctrl, std::optional<printer_info>)
+    -> caf::expected<std::function<void(chunk_ptr)>> override {
+    auto engine = engine::make_sink(args_);
+    if (not engine) {
+      return caf::make_error(ec::unspecified,
+                             fmt::format("failed to setup ZeroMQ: {}",
+                                         engine.error()));
+    }
+    return [&ctrl, engine = std::make_shared<class engine>(std::move(*engine))](
+             chunk_ptr chunk) mutable {
+      if (not chunk || chunk->size() == 0)
+        return;
+      if (auto error = engine->send(chunk))
+        ctrl.abort(error);
+    };
+  }
+
+  auto name() const -> std::string override {
+    return "zmq";
+  }
+
+  auto default_printer() const -> std::string override {
+    return "json";
+  }
+
+  auto is_joining() const -> bool override {
+    return true;
+  }
+
+  friend auto inspect(auto& f, zmq_saver& x) -> bool {
+    return f.object(x).pretty_name("zmq_saver").fields(f.field("args", x.args_));
+  }
+
+private:
+  connector_args args_;
+};
+
+class plugin final : public virtual loader_plugin<zmq_loader>,
+                     public virtual saver_plugin<zmq_saver> {
+public:
+  auto parse_loader(parser_interface& p) const
+    -> std::unique_ptr<plugin_loader> override {
+    auto args = connector_args{};
+    parse(p, args);
+    return std::make_unique<zmq_loader>(std::move(args));
+  }
+
+  auto parse_saver(parser_interface& p) const
+    -> std::unique_ptr<plugin_saver> override {
+    auto args = connector_args{};
+    parse(p, args);
+    return std::make_unique<zmq_saver>(std::move(args));
+  }
+
+  auto name() const -> std::string override {
+    return "zmq";
+  }
+
+private:
+  static void parse(parser_interface& p, connector_args& args) {
+    auto parser
+      = argument_parser{"zmq", "https://docs.tenzir.com/docs/connectors/zmq"};
+    auto endpoint = std::optional<located<std::string>>{};
+    parser.add(endpoint, "<endpoint>");
+    parser.add("-b,--bind", args.bind);
+    parser.add("-c,--connect", args.connect);
+    parser.add(endpoint, "<endpoint>");
```

Comment:
You add the positional argument to the parser twice here.

Reply:
Fixed in d7b5490d85.